server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -344,9 +346,18 @@ private PackageDefinition getPackageDefinition(String packageId) {
     }
 
     private PluggableSCMMaterialConfig toPluggableScmMaterialConfig(CRPluggableScmMaterial crPluggableScmMaterial) {
-        SCMs scms = getSCMs();
         String id = crPluggableScmMaterial.getScmId();
```

Comment:
`id` can be null now, right?
It is `scm_id`, which in [this example](https://github.com/gocd/gocd/issues/5694#issuecomment-454978548) can be skipped.

Reply:
On a second thought, if we allow `scm_id` to be not specified by user then we have a problem, because it is still needed to identify the scm at global level. It is used for example in api responses.
Do we generate scm_id then? Must be globally unique and I guess it should not change on each merge.

Reply:
Correct. It should be generated based on a hash of the fingerprint or something of that sort, I think. Maybe it's not too much of a problem if it changes on each merge, since it's only used for a ref - but it depends on the implementation. Ideally, it doesn't change - just makes it simpler.

Reply:
Yes, but only if it's creating a new SCM. Plugin configuration is required to create a new SCM

If they're creating a new SCM, then they can choose to provide an `scm_id` if they want. (See [here](https://github.com/gocd/gocd/pull/5756/files#diff-5236c1c9c0b120c4ff0f543e908fd396R356)). If they don't provide an `scm_id` then the post constructor callback will set it for them ([post constructor](https://github.com/gocd/gocd/blob/abe03ce0f013ff18b766e2378de711f22cbaa2d0/config/config-api/src/main/java/com/thoughtworks/go/domain/scm/SCM.java#L328))